src/stream/src/executor/project.rs

```diff
@@ -43,14 +48,17 @@ struct Inner {
     watermark_derivations: MultiMap<usize, usize>,
     /// Indices of nondecreasing expressions in the expression list.
     nondecreasing_expr_indices: Vec<usize>,
-    /// Last seen values of nondecreasing expressions, buffered to periodically produce watermarks.
-    last_nondec_expr_values: Vec<Option<ScalarImpl>>,
 
     /// the selectivity threshold which should be in `[0,1]`. for the chunk with selectivity less
     /// than the threshold, the Project executor will construct a new chunk before expr evaluation,
     materialize_selectivity_threshold: f64,
 }
 
+struct InnerState {
```

Comment:
We use `ExecutionVars` for this kind of struct before, in which the `vars` shows the mutability very well but is never associated with *state table* which stores executor's (persistent) internal **state**. I personally think the word `state` is having too many meanings in our system and may cause misunderstanding :(